internal/nodes/avi_model_l4_translator.go

```diff
@@ -57,9 +64,21 @@ func (o *AviObjectGraph) ConstructAviL4VsNode(svcObj *corev1.Service, key string
 		// subDomains[0] would either have the defaultSubDomain value
 		// or would default to the first dns subdomain it gets from the dns profile
 		if strings.HasPrefix(subDomains[0], ".") {
-			fqdn = svcObj.ObjectMeta.Name + "." + svcObj.ObjectMeta.Namespace + subDomains[0]
+			if lib.GetL4FqdnFormat() == 1 {
+				// Generate the FQDN based on the logic: <svc_name>.<namespace>.<sub-domain>
+				fqdn = svcObj.ObjectMeta.Name + "." + svcObj.ObjectMeta.Namespace + subDomains[0]
+			} else if lib.GetL4FqdnFormat() == 2 {
+				// Generate the FQDN based on the logic: <svc_name>-<namespace>.<sub-domain>
+				fqdn = svcObj.ObjectMeta.Name + "-" + svcObj.ObjectMeta.Namespace + subDomains[0]
+			}
```

Comment:
Can we move the above block to a separate functon to generate the fqdn and just call that function here and the below section to avoid duplication?

Reply:
I have optimised the code now.